natscontext/context.go

```diff
@@ -351,6 +352,15 @@ func (c *Context) NATSOptions(opts ...nats.Option) ([]nats.Option, error) {
 		nopts = append(nopts, nats.CustomInboxPrefix(c.InboxPrefix()))
 	}
 
+	u, err := url.Parse(c.ServerURL())
+	if err != nil {
+		return nil, err
+	}
+
+	if u.Path != "" {
+		nopts = append(nopts, nats.ProxyPath(u.Path))
```

Comment:
not sure this will work, its often to just put like `localhost` or `demo.nats.io` as server url.  And url.Parse will then do this:

```
&url.URL{Scheme:"", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"foo.com", RawPath:"", OmitHost:false, ForceQuery:false, RawQuery:"", Fragment:"", RawFragment:""}
```

ie. it doesnt know its a host it thinks its a path

Reply:
I see, nice catch :). I adjusted it so it will only take `proxyPath` when the URL is absolute. Let me know if this is ok on your end.